entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -290,19 +295,128 @@ private void doUpdate(RequestContext requestContext, EntityUpdateRequest request
           EntityQueryConverter.convertToAttributeValue(setAttribute.getValue()).build();
       String jsonValue = DocStoreJsonFormat.printer().print(attributeValue);
 
+      Map<Key, Map<String, Document>> entitiesUpdateMap = new HashMap<>();
+
       for (String entityId : request.getEntityIdsList()) {
         SingleValueKey key =
             new SingleValueKey(requestContext.getTenantId().orElseThrow(), entityId);
         // TODO better error reporting once doc store exposes the,
-        if (!entitiesCollection.updateSubDoc(key, subDocPath, new JSONDocument(jsonValue))) {
-          LOG.warn(
-              "Failed to update entity {}, subDocPath {}, with new doc {}.",
-              key,
-              subDocPath,
-              jsonValue);
+        if (entitiesUpdateMap.containsKey(key)) {
+          entitiesUpdateMap.get(key).put(subDocPath, new JSONDocument(jsonValue));
+        } else {
+          Map<String, Document> subDocument = new HashMap<>();
+          subDocument.put(subDocPath, new JSONDocument(jsonValue));
+          entitiesUpdateMap.put(key, subDocument);
         }
       }
+      try {
+        entitiesCollection.bulkUpdateSubDocs(entitiesUpdateMap);
```

Comment:
Couple of reasons

1. The update method currently only returns the number of successfully updated documents. https://github.com/hypertrace/document-store/blob/main/document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java#L47

We just tried to follow what the existing `bulkUpdate` function does for entities https://github.com/hypertrace/document-store/blob/main/document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java#L122

2. Trying to follow what the existing `update` API does. I think the use case was just to avoid another GET entities query to get the projected selections https://github.com/hypertrace/entity-service/blob/main/entity-service-api/src/main/proto/org/hypertrace/entity/query/service/v1/entity_query_request.proto#L25

Reply:
I'm really asking why `bulkUpdateSubDocs` doesn't, since it looks like at least here we need to do the following select/get anyway. Is it that the underlying store doesn't return this anyway, so even if we returned it from the api we're making two calls, an update and a select?

Reply:
need to check the update support in the underlying store, but shouldn't the `bulkUpdateSubDocs` follow the same sematincs as `bulkUpdate`?

Reply:
Yes, would make the same argument there - they both return the same object, so extending that object with the actual updated documents would affect both APIs.

Reply:
Yeah, we could extend that to return the actual updated documents, but I don't think we would be able to get the projected entities with just the requested selections in that case, unless we actually pass the set of requested projections to the document store API, and then ask it to return the projected documents (_as part of the same request, if Mongo/Postgres can support that_)

Reply:
In fact, these are 2 APIs merged into one
- bulk update entities
- get projected entities (_irrespective of the updated ones_)

now, that I think about it, the API sounds weird

Reply:
So taking a glance at the impls, at least postgres (not sure about mongo) does not support this, so it would probably make sense to keep it as a distinct API - that decision should likely bubble up then so we're consistent - if entity service is forced to make two calls, then we may as well have the caller make the two calls since there are certainly cases where we don't care about the updated docs.

Reply:
agreed. 

Mongo does support that, but I would rather prefer to drop the projections support from the API, as you mentioned too

@ankitchoudhary111 Can you please get rid of the projections support from the bulk update API?